### PR TITLE
BACKLOG-20295 : allow to override the mode where the sections are dis…

### PR DESCRIPTION
**Pull request by jsinovassin**
…played

https://jira.jahia.org/browse/BACKLOG-20295

### CODE REVIEW

File: src/test/resources/META-INF/jahia-content-editor-forms/forms/nt_base.override.displayModes.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "nodeType": "nt:base",
+  "hasPreview": true,
+  "priority": 2.0,
+  "sections": [
+    {
+      "name": "layout",
+      "displayModes": ["edit"]
```
Comment (tdraier) [inline src/test/resources/META-INF/jahia-content-editor-forms/forms/nt_base.override.displayModes.json:8]: why should we have "layout" section disabled by default for "create" ?

File: src/main/java/org/jahia/modules/contenteditor/api/forms/impl/EditorFormServiceImpl.java [hunk 2]
```diff
@@ -334,6 +341,11 @@ private EditorForm getEditorForm(ExtendedNodeType primaryNodeType, Locale uiLoca
         }
     }
 
+    private boolean shouldHideSection(EditorFormSectionDefinition section, JCRNodeWrapper existingNode){
+        String currentMode = existingNode == null ? "create" : "edit";
```
Comment (dgriffon) [inline src/main/java/org/jahia/modules/contenteditor/api/forms/impl/EditorFormServiceImpl.java:345]: Can you add a variable in some way to avoid to do the same test a line 301 `existingNode == null`, then pass the mode directly to the function (instead of the node)

Comment (jsinovassin) [reply src/main/java/org/jahia/modules/contenteditor/api/forms/impl/EditorFormServiceImpl.java]: Done!
